### DISCUSSION TITLE
Dynamically change the body of the request depending on dashboard variables

### DISCUSSION THREAD

**Discussion by Romathonat** [grafana/grafana-json-datasource#248] [Ideas]
Hello, and thank you for your excellent work.

Is it possible to dynamically change the body of the request depending on dashboard variables ?
I have cases where depending on the value of a textbox (if it is empty or not), the query I have to send to the backend needs to be changed (elasticsearch clause needs to be removed).

Thank you.

EDIT: Another solution could be to send different queries depending on a grafana variable (but it would still require a condition test).

**Comment by marcusolsson** [comment]
Thank you for the kind words, and for your suggestion on how to improve the plugin! ❤️ 

It sounds like this would require adding support for a more advanced templating language, such as [Handlebars](https://handlebarsjs.com/). I don't think it would have that big of an impact on UX (a checkbox to enable templating?), but it would require another non-trivial dependency which I'm hesitant to add without feedback from more users.

**Comment by Romathonat** [comment]
Yes I understand your point of view.
For those who encouter the same problem as me, we can handle it with mustache in ElasticSearch: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html#_conditional_clauses

However we have to escape the request as a string (it is no longer a valid JSON), making it very difficult to maintain/debug.

**Comment by marcusolsson** [comment]
Adding this to the backlog as there seems to be some support for this feature.

**Comment by faedrielle** [comment]
Hello and thanks for the wonderful work!

I'd like to bump this issue, since in my case, I need to send timestamps (taken from the Time picker) in the body of the request. In SQL, this would be done with $__timeFrom(), but I haven't been able to do this in the plugin.

Simplified request example:
POST /api/Data
Body: {"since": "2021-03-01T00:00:00.000Z", "till": "2021-03-01T17:41:44.3333333Z"}

Cheers!

**Comment by Dexter-14** [comment]
So do we have a way for this , I am using Json api plugin and want to change the body according to the field variables.

**Comment by maximus1127** [comment]
Not sure if anyone is still following this issue, but here is how to dynamically change body variables:

![image](https://github.com/grafana/grafana-json-datasource/assets/31026284/32d1d0ed-8c1e-46e4-8fd1-11295d6527e2)
